File: Arrays_Hashing/TwoSum/TwoSum.kt
class Solution {
    fun twoSum(nums: IntArray, target: Int): IntArray {
        val map = HashMap<Int, Int>()
        for (i in 0..<nums.size) {
            map[nums[i]] = i
        }

        for(i in 0..<nums.size) {
            val complement = target - nums[i]
            if (map[complement] != null && map[complement] != i) {
                return intArrayOf(i, map[complement]!!)
            }
        }

        return intArrayOf(0,0)
    }
}

File: Arrays_Hashing/TwoSum/TwoSum.swift
class Solution {
    func twoSum(_ nums: [Int], _ target: Int) -> [Int] {
        var dict: [Int: Int] = [:]
        for i in 0..<nums.count {
            dict[nums[i]] = i
        }
        
        for j in 0..<nums.count {
            if let idx = dict[target - nums[j]], idx != j {
                return [dict[target - nums[j]]!, j]
            }
        }
        
         return [-1,-1]
    }
}
